opendevin/runtime/client/client.py

```diff
@@ -1,229 +1,150 @@
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(server)
-        loop.run_forever()
-    
-    def init_shell(self) -> None:
-        # run as root
-        self.shell = pexpect.spawn('/bin/bash', encoding='utf-8')
-        self.shell.expect(r'[$#] ')
+app = FastAPI()
 
-    async def listen(self, websocket):
-        try:
-            async for message in websocket:
-                event_str = json.loads(message)
-                event = event_from_dict(event_str)
-                if isinstance(event, Action):
-                    observation = self.run_action(event)
-                    await websocket.send(json.dumps(event_to_dict(observation)))
-        except ConnectionClosed:
-            print("Connection closed")
-    
-    def run_action(self, action) -> Observation:
-        # Should only receive Action CmdRunAction and IPythonRunCellAction
-        action_type = action.action  # type: ignore[attr-defined]
-        observation = getattr(self, action_type)(action)
-        # TODO: see comments in https://github.com/OpenDevin/OpenDevin/pull/2603#discussion_r1668994137
-        observation._parent = action.id  # type: ignore[attr-defined]
-        return observation
-    
-    def run(self, action: CmdRunAction) -> Observation:
-        return self._run_command(action.command)
-    
-    def _run_command(self, command: str) -> Observation:
-        try:
-            output, exit_code = self.execute(command)
-            return CmdOutputObservation(
-                command_id=-1, content=str(output), command=command, exit_code=exit_code
-            )
-        except UnicodeDecodeError:
-            return ErrorObservation('Command output could not be decoded as utf-8')
 
-    def clean_up(self,input_text):
-        # Remove escape sequences
-        cleaned_text = re.sub(r'\x1b\[[0-9;?]*[a-zA-Z]', '', input_text)
-        # Remove carriage returns and other control characters
-        cleaned_text = re.sub(r'[\r\n\t]', '', cleaned_text)
-        return cleaned_text
+class ActionRequest(BaseModel):
+    action: dict
+
+
+class RuntimeClient:
+    """RuntimeClient is running inside docker sandbox.
+    It is responsible for executing actions received from OpenDevin backend and producing observations.
+    """
+
+    def __init__(self, plugins_to_load: list[Plugin]) -> None:
+        self._init_bash_shell()
+        self.lock = asyncio.Lock()
+        self.plugins: dict[str, Plugin] = {}
 
-    def execute(self, command):
-        print(f"Received command: {command}")
+        for plugin in plugins_to_load:
+            plugin.initialize()
+            self.plugins[plugin.name] = plugin
+            logger.info(f'Initializing plugin: {plugin.name}')
+
+    def _init_bash_shell(self) -> None:
+        self.shell = pexpect.spawn('/bin/bash', encoding='utf-8', echo=False)
+        self.shell.expect(r'[$#] ')
+
+    def _execute_bash(self, command):
+        logger.info(f'Received command: {command}')
```

Comment:
Because client run inside the docker, I am thinking where is the output of `logger.info`.

Reply:
You can check it using `docker logs CONTAINER NAME` - but i guess this is only for dev developing purpose.